e2e/playground.py

```diff
@@ -76,15 +79,43 @@
 
 OptionalStr = typing.Optional[str]
 OptionalInt = typing.Optional[int]
+OptionalFloat = typing.Optional[float]
+OptionalBool = typing.Optional[bool]
 
 PatternOrStr = typing.Union[str, typing.Pattern[str]]
-TextValue = typing.Union[PatternOrStr, None]
 AttrValue = typing.Union[PatternOrStr, None]
 StyleValue = typing.Union[PatternOrStr, None]
 
 Timeout = typing.Union[float, None]
 InitLocator = typing.Union[Locator, str]
 
+R = typing.TypeVar("R")
+M1 = typing.TypeVar("M1")
+M2 = typing.TypeVar("M2")
+
+
+def is_missing(x: typing.Any) -> typing.TypeGuard[MISSING_TYPE]:
+    return isinstance(x, MISSING_TYPE)
+
+
+def not_is_missing(x: typing.Union[R, MISSING_TYPE]) -> typing.TypeGuard[R]:
```

Comment:
Can you make a comment about the typeguard/narrowing thing that makes this necessary?